src/node_contextify.cc

```diff
@@ -132,41 +133,51 @@ class ContextifyContext {
-        //
-        // However, this doesn't handle ES6-style properties configured with
-        // Object.defineProperty, and that's exactly what we're up against at
-        // this point.  ForceSet(key,val,att) only supports value properties
-        // with the ES3-style attribute flags (DontDelete/DontEnum/ReadOnly),
-        // which doesn't faithfully capture the full range of configurations
-        // that can be done using Object.defineProperty.
-        if (clone_property_method.IsEmpty()) {
-          Local<String> code = FIXED_ONE_BYTE_STRING(env()->isolate(),
-              "(function cloneProperty(source, key, target) {\n"
-              "  if (key === 'Proxy') return;\n"
-              "  try {\n"
-              "    var desc = Object.getOwnPropertyDescriptor(source, key);\n"
-              "    if (desc.value === source) desc.value = target;\n"
-              "    Object.defineProperty(target, key, desc);\n"
-              "  } catch (e) {\n"
-              "   // Catch sealed properties errors\n"
-              "  }\n"
-              "})");
-
-          Local<Script> script =
-              Script::Compile(context, code).ToLocalChecked();
-          clone_property_method = Local<Function>::Cast(script->Run());
-          CHECK(clone_property_method->IsFunction());
+        Local<Object> desc_from_vm_context =
+            global->GetOwnPropertyDescriptor(context, key)
+            .ToLocalChecked().As<Object>();
+
+        bool isAccessor = desc_from_vm_context
+            ->Has(context, env()->get_string())
+            .FromJust() || desc_from_vm_context
+                ->Has(context, env()->set_string())
+                .FromJust();
+
+        PropertyDescriptor* desc;
+
+        if (!isAccessor) {
+          Local<Value> value = desc_from_vm_context
+              ->Get(context, env()->value_string()).ToLocalChecked();
+
+          bool writable = desc_from_vm_context
+              ->Get(context, env()->writable_string()).ToLocalChecked()
+              .As<Value>()->BooleanValue();
+
+          desc = new PropertyDescriptor(value, writable);
+        } else {
+          Local<Function> get = desc_from_vm_context
+              ->Get(context, env()->get_string())
+              .ToLocalChecked().As<Function>();
+          Local<Function> set = desc_from_vm_context
+              ->Get(context, env()->set_string())
+              .ToLocalChecked().As<Function>();
+
+          desc = new PropertyDescriptor(get, set);
         }
-        Local<Value> args[] = { global, key, sandbox_obj };
-        clone_property_method->Call(global, arraysize(args), args);
-      }
+
+        desc->set_configurable(desc_from_vm_context
+            ->Get(context, env()->configurable_string()).ToLocalChecked()
+            .As<Value>()->BooleanValue());
+        desc->set_enumerable(desc_from_vm_context
+            ->Get(context, env()->enumerable_string()).ToLocalChecked()
+            .As<Value>()->BooleanValue());
```

Comment:
You can get rid of the `desc = new PropertyDescriptor(...)` and `delete desc` calls if you wrap this code in a lambda, like this:
```c++
auto configure = [&] (PropertyDescriptor* desc) {
  // ...
};
```
Which you then call like this:
```c++
if (is_accessor) {
  // ...
  PropertyDescriptor desc(get, set);
  configure(&desc);
} else {
  // ...
  PropertyDescriptor desc(value, writable);
  configure(&desc);
}
```